src/test/icalparser_ctrl_test.c

```diff
@@ -0,0 +1,81 @@
+/*======================================================================
+ FILE: icalparser_ctrl_test.c
+ CREATOR: Robert Stepanek, 10 October 2023
+
+ SPDX-FileCopyrightText: 2000 Eric Busboom <eric@civicknowledge.com>
+
+ SPDX-License-Identifier: LGPL-2.1-only OR MPL-2.0
+
+======================================================================*/
+
+/*
+ * Program to test handling of CONTROL characters in iCalendar parsing.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include "libical/ical.h"
+
+static void assert_ctrl(enum icalparser_ctrl ctrl,
+                        const char *data,
+                        const char *want_desc,
+                        int want_xlicerror)
+{
+    icalparser_set_ctrl(ctrl);
+
+    icalcomponent *ical = icalparser_parse_string(data);
+    assert(ical);
+
+    icalcomponent *comp = icalcomponent_get_first_real_component(ical);
```

Comment:
This produces a compiler warning:
```
[50/57] Building C object src/test/CMakeFiles/parser_ctrl.dir/icalparser_ctrl_test.c.o                                                  
/git/libical/src/test/icalparser_ctrl_test.c: In function ‘assert_ctrl’:                                                                
/git/libical/src/test/icalparser_ctrl_test.c:31:20: warning: unused variable ‘comp’ [-Wunused-variable]
   31 |     icalcomponent *comp = icalcomponent_get_first_real_component(ical);                                                         
      |                    ^~~~                                                                                                         
```
Consider rewriting as:
```c
assert(icalcomponent_get_first_real_component(ical))
```

Reply:
I'm not following. We are using `comp` in later lines in this function and my compiler rightly so complains if I apply your suggested patch

```
libical/src/test/icalparser_ctrl_test.c:34:65: error: ‘comp’ undeclared (first use in this function)
   34 |         assert(!strcmp(want_desc, icalcomponent_get_description(comp)));
```

Reply:
On the one side you are right, that this is not the correct fix.

On the other side there is a compiler warning, when the content of `assert()`s is stripped by the preprocessor and is invisible to the compiler.

Reply:
This is test code, and other test harnesses in the same directory make use of assert, too. I think it's safe to assume that compiling it when stripping asserts is an error. If asserts should be replaced with something libical-test specific, then it would make sense to do that in general for the test code.

Reply:
When I compile libical this is the single compiler warning I get.